Tests/CodeEditSourceEditorTests/Controller/TextViewController+IndentTests.swift

```diff
@@ -21,73 +23,102 @@ final class TextViewControllerIndentTests: XCTestCase {
         controller.handleIndent(inwards: true)
 
-        XCTAssertEqual(controller.string, "This is a test string")
+        expectNoDifference(controller.string, "This is a test string")
 
         // Normally, 4 spaces are used for indentation; however, now we only insert 2 leading spaces.
         // The outcome should be the same, though.
         controller.setText("  This is a test string")
         controller.cursorPositions = cursorPositions
+        controller.textView.selectionManager.textSelections = [.init(range: NSRange(location: 0, length: 0))]
         controller.handleIndent(inwards: true)
 
-        XCTAssertEqual(controller.string, "This is a test string")
+        expectNoDifference(controller.string, "This is a test string")
     }
 
     func testHandleIndentWithSpacesOutwards() {
         controller.setText("This is a test string")
         let cursorPositions = [CursorPosition(range: NSRange(location: 0, length: 0))]
+        controller.textView.selectionManager.textSelections = [.init(range: NSRange(location: 0, length: 0))]
         controller.cursorPositions = cursorPositions
 
         controller.handleIndent(inwards: false)
 
-        XCTAssertEqual(controller.string, "    This is a test string")
+        expectNoDifference(controller.string, "    This is a test string")
     }
 
     func testHandleIndentWithTabsInwards() {
         controller.setText("\tThis is a test string")
         controller.indentOption = .tab
         let cursorPositions = [CursorPosition(range: NSRange(location: 0, length: 0))]
+        controller.textView.selectionManager.textSelections = [.init(range: NSRange(location: 0, length: 0))]
         controller.cursorPositions = cursorPositions
 
         controller.handleIndent(inwards: true)
 
-        XCTAssertEqual(controller.string, "This is a test string")
+        expectNoDifference(controller.string, "This is a test string")
     }
 
     func testHandleIndentWithTabsOutwards() {
         controller.setText("This is a test string")
         controller.indentOption = .tab
         let cursorPositions = [CursorPosition(range: NSRange(location: 0, length: 0))]
+        controller.textView.selectionManager.textSelections = [.init(range: NSRange(location: 0, length: 0))]
         controller.cursorPositions = cursorPositions
 
         controller.handleIndent()
 
         // Normally, we expect nothing to happen because only one line is selected.
         // However, this logic is not handled inside `handleIndent`.
-        XCTAssertEqual(controller.string, "\tThis is a test string")
+        expectNoDifference(controller.string, "\tThis is a test string")
     }
 
     func testHandleIndentMultiLine() {
         controller.indentOption = .tab
-        controller.setText("This is a test string\nWith multiple lines\nAnd some indentation")
-        let cursorPositions = [CursorPosition(range: NSRange(location: 0, length: 5))]
+        let strings: [(NSString, Int)] = [
```

Comment:
Because `handleIndent` now relies on a selection which relies on the red/black tree. I needed to add different sets of characters to get the tests passing (low key an integration test but that's ok).